### PR TITLE
[BEAM-1381] Adding metrics() to DataflowPipelineResult.

### PR DESCRIPTION
**Pull request by pabloem**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by pabloem** [comment]
r: @aaltay

File: sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+"""
+DataflowRunner implementation of MetricResults. It is in charge of
+responding to queries of current metrics by going to the dataflow
+service.
+"""
+
+from apache_beam.metrics.metric import MetricResults
+
+
+#TODO(pabloem) Implement this once metrics are queriable from dataflow service
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py]: Add a JIRA issue here and to the PR title.

File: sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class DataflowMetrics(MetricResults):
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py]: Do you need an equivalent of `DirectMetric`, or `DirectMetric` is an implementation detail?

File: sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py]: Add a 'dataflow_metrics_test.py' that we can populate later.

File: sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py (hunk 2 above)
Comment (pabloem) [inline sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py]: That's an implementation detail. That functionality is done in the dataflow back end in our case.

File: sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py (hunk 1 above)
Comment (pabloem) [inline sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py]: Added.

File: sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py (hunk 3 above)
Comment (pabloem) [inline sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py]: Done.

**Comment by pabloem** [comment]
Done.

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**